exist-core/src/test/xquery/filter-function-items-in-context.xql

```diff
@@ -0,0 +1,69 @@
+xquery version "3.1";
+
+(:~
+ : tests focussing on filtering function items in context
+ :)
+module namespace filter-function-items-in-context="http://exist-db.org/xquery/test/filter-function-items-in-context";
+
+
+declare namespace test="http://exist-db.org/xquery/xqsuite";
+
+
+declare variable $filter-function-items-in-context:maps := ( map { "id": 1 }, map { "id": 2, "foo": 3 } );
+declare variable $filter-function-items-in-context:arrs := ( [1, 2], ["a", "b", "c"] );
+
+declare
+    %test:assertEquals(1)
+function filter-function-items-in-context:map-function() {
+    $filter-function-items-in-context:maps[map:size(.) eq 1]?id
+};
+
+declare
+    %test:assertEquals(2)
+function filter-function-items-in-context:map-lookup() {
+    $filter-function-items-in-context:maps[map:get(., "id") eq 2]?id
+};
+
+declare
+    %test:assertEquals("a", "b", "c")
+function filter-function-items-in-context:array-lookup() {
+    $filter-function-items-in-context:arrs[array:get(.,2) instance of xs:string]?*
+};
+
+declare
+    %test:assertEquals(1)
+function filter-function-items-in-context:array-function() {
+    $filter-function-items-in-context:arrs[array:size(.) eq 2]?1
+};
+
+declare
+    %test:assertEquals(1)
+function filter-function-items-in-context:named-function-arity () {
+    (
+        current-date#0,
+        string-join#1
+    )
+    [function-arity(.) eq 1] => count()
+};
+
+declare
+    %test:assertEquals(1)
+function filter-function-items-in-context:anonymous-function-arity () {
+    (
+        function() { "foo" },
+        function($x) { "bar" }
+    )
+    [function-arity(.) eq 1] => count()
+};
+
+declare
+    %test:assertEquals(4)
+function filter-function-items-in-context:mixed-sequence () {
+    (
+        $filter-function-items-in-context:arrs,
+        $filter-function-items-in-context:maps,
+        string-join#1,
+        function($x) { $x }
+    )
+    [. instance of function(*)] => count()
+};
```

Comment:
`([1,2],[1,2,3])[.?2 eq 2]` will be added, but will fail ATM

Reply:
How about `([1,2],[1,2,3])[?2 eq 2]`? I'm curious about how `.?` differs from `?` - is the former redundant?

Reply:
Of course these need to be added as well. But since the unary lookup could be seen as a different construct and is also a different path in the parser, I do not know if it should be part of this specific test suite.

Reply:
done